staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -786,3 +806,7 @@ func appendListItem(v reflect.Value, data []byte, rev uint64, pred storage.Selec
 func notFound(key string) clientv3.Cmp {
 	return clientv3.Compare(clientv3.ModRevision(key), "=", 0)
 }
+
+func getTypeName(obj interface{}) string {
```

Comment:
This might be ok for a first pass, but this won't correctly subdivide metrics for custom resources (CRDs) and could potentially have a skewed latency distribution for a few of our special case resources (the masters create a store and write leased endpoint objects to maintain the `kubernetes` service, we reuse locks in a few places, etc)

However, it's probably good enough for now.  Can you add godoc to this function indicating both of the exceptions above and indicate that a future revision should require the store get passed enough info to fill this better?

Reply:
ok

Reply:
strip the `get` prefix. `typeName(obj)` is fine and more Go-like.

Reply:
not fixed